llvm/lib/Target/AIE/AIE2InstructionSelector.cpp

```diff
@@ -4427,51 +4427,57 @@ bool AIE2InstructionSelector::select512BitG_AIE_LOAD_STORE(
   }
 }
 
-std::optional<LoadStoreOpcodes>
-getCombinedOpcodeCONVLoad(const MachineInstr &MemOp, const MachineInstr &CombOp,
-                          std::optional<APInt> Immediate) {
-
-  // const bool AlwaysFitsImmediateRange = true;
-  const bool NoImmediate = false;
-  if (CombOp.getOpcode() != AIE2::G_INTRINSIC ||
-      cast<GIntrinsic>(CombOp).getIntrinsicID() !=
-          Intrinsic::aie2_v16bf16_to_v16accfloat)
-    return {};
-
+bool getVLDA_CONVOpcode(const MachineInstr &MemOp,
+                        const std::optional<APInt> Immediate,
+                        unsigned &ISelOpcode, bool &FitsImmediateRange) {
   if (!MemOp.mayLoad())
-    return {};
-
-  assert(getLoadStoreSize(MemOp) == 256 && "Unexpected VLDA.CONV size");
```

Comment:
As I understood, this assert is the origin of the problem. If we move it outside this logic everything goes well. `getLoadStoreSize` expects LD/ST opcodes, anything else will crash. Your PR basically moved this assert to a safe place. Looks good to me.

Reply:
Right..